app/models/condition.rb

```diff
@@ -19,10 +19,11 @@ class Condition < ApplicationRecord
   validates :questioning, presence: true
 
   delegate :has_options?, :full_dotted_rank, to: :ref_qing, prefix: true
-  delegate :form, :form_id, to: :questioning
+  delegate :form, :form_id, :refable_qings, to: :questioning
 
   scope :referring_to_question, ->(q) { where(ref_qing_id: q.qing_ids) }
   scope :with_display_role, -> { where(role: "display") }
+  scope :by_ref_qing_rank, -> { joins(:ref_qing).order("form_items.rank") }
```

Comment:
Update: Is it for spec determinism? If so, I guess it's not perfect but it helps? Could you put a comment? Otherwise it's not clear what the purpose is. Thx.

Reply:
It's also for some determinism in the hint order on the form screen. We really need to find a way to introduce some way of querying a sortable complete rank at the DB level -- there are a lot of operations that are complex or slow because we don't have this capability.

Reply:
Agreed